src/plugins/score-plugin-remotecontrol/RemoteControl/Http_server.cpp

```diff
@@ -0,0 +1,325 @@
+{
+    bool close = false;
+    beast::error_code ec;
+
+    // This buffer is required to persist across reads
+    beast::flat_buffer buffer;
+
+    // This lambda is used to send messages
+    send_lambda<tcp::socket> lambda{socket, close, ec};
+
+    for(;;)
+    {
+        // Read a request
+        http::request<http::string_body> req;
+        http::read(socket, buffer, req, ec);
+        if(ec == http::error::end_of_stream)
+            break;
+        if(ec)
+            return Http_server::fail(ec, "read");
+
+        // Send the response
+        Http_server::handle_request(*doc_root, std::move(req), lambda);
+        if(ec)
+            return Http_server::fail(ec, "write");
+        if(close)
+        {
+            // This means we should close the connection, usually because
+            // the response indicated the "Connection: close" semantic.
+            break;
+        }
+    }
+
+    // Send a TCP shutdown
+    socket.shutdown(tcp::socket::shutdown_send, ec);
+
+    // At this point the connection is closed gracefully
+}
+
+//------------------------------------------------------------------------------
+
+// Set the IP address in the remote.html file
+void
+Http_server::set_ip_address(std::string address)
+{
+    std::rename("./src/plugins/score-plugin-remotecontrol/CMakeFiles/score_plugin_remotecontrol.dir/RemoteControl/build-wasm/remote.html",
+                "./src/plugins/score-plugin-remotecontrol/CMakeFiles/score_plugin_remotecontrol.dir/RemoteControl/build-wasm/remote.html~");
+
+    std::ifstream old_file("./src/plugins/score-plugin-remotecontrol/CMakeFiles/score_plugin_remotecontrol.dir/RemoteControl/build-wasm/remote.html~");
+    std::ofstream new_file("./src/plugins/score-plugin-remotecontrol/CMakeFiles/score_plugin_remotecontrol.dir/RemoteControl/build-wasm/remote.html");
+
+    std::string addr = "\"" + address + "\"";
+
+    for( std::string contents_of_file; std::getline(old_file, contents_of_file); ) {
+      std::string::size_type position = contents_of_file.find("%SCORE_IP_ADDRESS%");
+      if( position != std::string::npos )
+        contents_of_file = contents_of_file.replace(position, 18, addr);
+      new_file << contents_of_file << '\n';
+    }
+}
+
+//------------------------------------------------------------------------------
+
+// Launch the open_server function in a thread
+void
+Http_server::start_thread()
+{
+    m_serverThread = std::thread{[this] { open_server(); }};
+}
+
+//------------------------------------------------------------------------------
+
+// Open a server using sockets
+int
+Http_server::open_server()
+{
+    try
+    {
+        auto const address2 = net::ip::make_address("0.0.0.0");
+        auto const port = static_cast<unsigned short>(std::atoi("8080"));
+        auto const m_docRoot = std::make_shared<std::string>("./src/plugins/score-plugin-remotecontrol/CMakeFiles/score_plugin_remotecontrol.dir/RemoteControl/build-wasm/");
```

Comment:
this line only works on the build computer.
instead we have to look for it in the user library which is where it will be installed on the user's computer:


```
#include <Library/LibrarySettings.hpp>
#include <score/application/ApplicationContext.hpp>

QString chemin_vers_dossier_packages = score::AppContext().settings<Library::Settings::Model>().getPackagesPath();
```